### PR TITLE
DBAAS-69: watch DBaaSProvider inventory and connection CRs 

### PR DESCRIPTION
**Pull request by xieshenzh**
## Description

Watch the provider ConfigMaps changes and re-deploy DBaaS operator
Jira: https://issues.redhat.com/browse/DBAAS-69

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist (Definition of Done)

- [ ] All acceptance criteria specified in JIRA or in issue number have been completed
- [ ] Unit tests added that prove the fix is effective or the feature works 
- [ ] Documentation added for the feature
- [ ] all relevant tests are passing
- [ ] Code Review completed
- [ ] Verified independently by reviewer

### CODE REVIEW

**Comment by jeremyary** [comment]
On hold pending some discussion here as we may have to change where we find those CM's: https://docs.google.com/document/d/16zmkLMFVj6l3_2P3c1Dlyu6C65Iwd5ZnXRKK60xxrLA/edit?disco=AAAANQtWYpI

**Comment by jeremyary** [comment]
Content of comment thread in previous link has been moved into its own section for discussion (outcome still pending):
https://docs.google.com/document/d/16zmkLMFVj6l3_2P3c1Dlyu6C65Iwd5ZnXRKK60xxrLA/edit#heading=h.z264ael09a0f

**Comment by jeremyary** [comment]
Removing hold, but affected by the DBaaSProvider refactor.

**Comment by xieshenzh** [comment]
After some investigation, I found `controller` supports [watching events from new sources after it is started](https://github.com/kubernetes-sigs/controller-runtime/blob/3203c07f930f40ff1ecbaf1bfaddfd9486c30ac1/pkg/internal/controller/controller.go#L122-L131). 
I have updated the DBaaSProvider controller to add new watches.

**Review by jeremyary — APPROVED** [review]

**Comment by jeremyary** [comment]
lgtm, IMO definitely feels more like it fits the controller paradigm this way, too. If you're good to drop the hold tag, I'll merge us up. Thanks for your efforts on this, I noticed what time this last change came in! :)

File: controllers/dbaasprovider_controller.go [hunk 1]
```diff
@@ -40,16 +46,52 @@ type DBaaSProviderReconciler struct {
...
+	logger.Info("Watching Provider Connection CR")
 
 	return ctrl.Result{}, nil
 }
 
 // SetupWithManager sets up the controller with the Manager.
 func (r *DBaaSProviderReconciler) SetupWithManager(mgr ctrl.Manager) error {
 	return ctrl.NewControllerManagedBy(mgr).
-		For(&dbaasv1alpha1.DBaaSProvider{}).
+		For(&v1alpha1.DBaaSProvider{}, builder.WithPredicates(filterEventPredicate)).
```
Comment (xieshenzh) [inline controllers/dbaasprovider_controller.go:80]: Made a small change here. Thanks.

Comment (jeremyary) [reply controllers/dbaasprovider_controller.go:80]: IIUC, you're just changing from a global predicate (`WithFilterEvent(...)`) to a specific predicate on the builder. We're using global predicates in our provider operator, so have you read something that says to prefer one over the other?

Comment (xieshenzh) [reply controllers/dbaasprovider_controller.go:80]: The `filterEventPredicate` only works with `DBaaSProvider` events. 
If the controller will watch events of other object types in the future, I don't think the `filterEventPredicate` should be applied. So I changed it from a global predicate.

Comment (jeremyary) [reply controllers/dbaasprovider_controller.go:80]: ok, makes sense. Just making sure we didn't need to consider for the others, will keep it in mind. Thanks!

**Review by jeremyary — APPROVED** [review]

**Pull request merged**